test_runner/src/main/kotlin/ftl/run/common/PollMatrices.kt

```diff
@@ -36,8 +37,15 @@ private suspend fun pollMatrix(matrixId: String, stopwatch: StopWatch, args: IAr
     var refreshedMatrix = GcTestMatrix.refresh(matrixId, args.project)
     val watch = StopWatchMatrix(stopwatch, matrixId)
     val runningDevices = RunningDevices(stopwatch, refreshedMatrix.testExecutions)
+    var printWebLink = true
 
     while (true) {
+        val webLink = refreshedMatrix.webLink()
+        if (printWebLink && webLink.isNotBlank()) {
+            printWebLink = false
+            println(webLink)
+        }
```

Comment:
Is there a cleaner way to print the web link once per matrix?

Reply:
I am wondering if we could print this link using this logic in `SavedMatrix`
https://github.com/Flank/flank/blob/d91f7784dadfafb9359a9e487d53507fda57e432/test_runner/src/main/kotlin/ftl/json/SavedMatrix.kt#L65-L67
AFAIK link is updated only once for each matrix (please confirm that, I've tested it with couple of examples but might be some corner cases about which I am not aware of).
Also, I think we could print it in more user friendly way --  with indention, message saying what exactly this is etc.
I am just thinking, let me know what you think.

Reply:
I like that idea!

My understanding is the matrix link is only updated once. I'll update the pull request.

Reply:
If we need to print web link ASAP "after starting run", I guess that the cleaner way is to do first refresh of matrices somewhere [here](https://github.com/Flank/flank/blob/f114c83789d95525b35d7daa6f200d7feebd23d6/test_runner/src/main/kotlin/ftl/run/platform/common/AfterRunTests.kt#L29). So the web link can be properly generated and printed before "PollMatrices" message will appears. But it's not good idea if number of additional pulls matters.